dev/io.openliberty.microprofile.telemetry.2.0.logging.internal/src/io/openliberty/microprofile/telemetry20/logging/internal/OpenTelemetryLogHandler.java

```diff
@@ -0,0 +1,279 @@
+	                    				otelInstance = OpenTelemetryAccessor.getOpenTelemetryInfo(appName).getOpenTelemetry();
+	                    			}
+	                    		}
+	                	}
+	            	}
+	        }
+	        
+	        if(otelInstance == null) {
+	        	otelInstance = this.openTelemetry;
+	        }
+	        
+	        if (otelInstance != null) {
+	            builder = otelInstance.getLogsBridge().loggerBuilder(OpenTelemetryConstants.INSTRUMENTATION_NAME).build().logRecordBuilder();
+	        }
+	        
+	        if(builder != null)
+	        	mapLibertyLogRecordToOTelLogRecord(builder, logData, eventType, attributes);
+	    }
+	    return builder;
+	}
+	
+	private void mapLibertyLogRecordToOTelLogRecord(LogRecordBuilder builder, LogTraceData logData, String eventType, AttributesBuilder attributes) {
+        boolean isMessageEvent = eventType.equals(CollectorConstants.MESSAGES_LOG_EVENT_TYPE);
+	    
+	    // Get message from LogData and set it in the LogRecordBuilder
+	    String message = logData.getMessage();
+	    builder.setBody(message);
+	    
+	    // Get Timestamp from LogData and set it in the LogRecordBuilder
+	    builder.setTimestamp(logData.getDatetime(), TimeUnit.MILLISECONDS);
+	    
+	    // Get Log Level from LogData and set it in the LogRecordBuilder
+	    String loglevel = logData.getLoglevel();
+	    builder.setSeverity(wsLevelToSeverity(loglevel));
+	    
+	    // Get Log Severity from LogData and set it in the LogRecordBuilder
+	    String logSeverity = logData.getSeverity();
+	    builder.setSeverityText(logSeverity);
+	    
+	    // Add Thread information to Attributes Builder
+	    attributes.put(SemanticAttributes.THREAD_NAME, logData.getThreadName());
+	    attributes.put(SemanticAttributes.THREAD_ID, logData.getThreadId());
+	    
+	    // Add Throwable information to Attribute Builder
+	    String exceptionName = logData.getExceptionName();
+	    String throwable = logData.getThrowable();
+	    if (throwable != null) {
+	    	    attributes.put(SemanticAttributes.EXCEPTION_MESSAGE, exceptionName);
+	    	    attributes.put(SemanticAttributes.EXCEPTION_STACKTRACE, throwable);
+	    }
+	    
+	    // Add additional log information from LogData to Attributes Builder
+	    attributes.put(LogTraceData.getModuleKey(0, isMessageEvent), logData.getModule())
+	              .put(LogTraceData.getMessageIdKey(0, isMessageEvent), logData.getMessageId())
+	              .put(LogTraceData.getMethodNameKey(0, isMessageEvent), logData.getMethodName())
+	              .put(LogTraceData.getClassNameKey(0, isMessageEvent), logData.getClassName())
+	              .put(LogTraceData.getSequenceKey(0, isMessageEvent), logData.getSequence());
+	    
+	    
+	    // Set the Attributes to the builder.
+	    builder.setAllAttributes(attributes.build());
+	    
+	    // span context
+	    //builder.setContext(Context.current());
```

Comment:
should we be using this?

Reply:
We still need to get the span/trace ids to correlate the logs. However, setting the context here, would be redundant, cause the wrong context will be set, since the collector collects all the logs in the buffer. The context will be different.